openhands/sdk/llm/utils/metrics.py

```diff
@@ -242,6 +255,31 @@ def log(self) -> str:
             logs += f"{key}: {value}\n"
         return logs
 
+    def __getstate__(self):
+        """Exclude the lock from being pickled/copied."""
+        state = self.__dict__.copy()
+        # Remove the lock from the state
+        state.pop("_lock", None)
+        return state
+
+    def __setstate__(self, state):
+        """Restore the object and create a new lock."""
+        self.__dict__.update(state)
+        self._lock = threading.Lock()
+
+    def __deepcopy__(self, memo=None):
+        """Custom deepcopy implementation that handles the threading lock."""
+        # Create a new instance using Pydantic's proper initialization
+        # but exclude the lock from the data
+        data = {}
+        for field_name in self.__class__.model_fields:
+            field_value = getattr(self, field_name)
+            data[field_name] = copy.deepcopy(field_value, memo)
+
+        # Create new instance with proper Pydantic initialization
+        new_instance = self.__class__(**data)
+        return new_instance
+
     def deep_copy(self) -> "Metrics":
         """Create a deep copy of the Metrics object."""
         return copy.deepcopy(self)
```

Comment:
@openhands can we do the following

1. get rid of `__deepcopy__`, `__getstate__`, `__setstate__`
2. make sure to rewrite `deep_copy` method so that we're using the build in pydantic method instead `self.model_copy(deep=True)`


When using the pydantic built in, make sure that the new copy has its own copy of a lock and that the private attributes (such as the lock) are not copied over or break the copying process